packages/@aws-cdk/aws-rds/test/instance.test.ts

```diff
@@ -1617,6 +1617,31 @@ describe('instance', () => {
       },
     });
   });
+
+  test('engine is specified for read replica using domain', () => {
+    // GIVEN
+    const instanceType = ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.SMALL);
+    const engine = rds.DatabaseInstanceEngine.postgres({ version: rds.PostgresEngineVersion.VER_13 });
+    const source = new rds.DatabaseInstance(stack, 'Source', {
+      engine,
+      instanceType,
+      vpc,
+    });
+
+    // WHEN
+    new rds.DatabaseInstanceReadReplica(stack, 'Replica', {
+      sourceDatabaseInstance: source,
+      instanceType,
+      vpc,
+      domain: 'my-domain',
+    });
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::RDS::DBInstance', {
+      SourceDBInstanceIdentifier: Match.anyValue(),
+      Engine: 'postgres',
+    });
+  });
```

Comment:
I wonder... since this is a CFN validation, do we need a new small integration test for this, to confirm it deploys?

Reply:
Ideally yes but it's complicated because it means we need to create a domain using L1 resources from `@aws-cdk/aws-directoryservice`...

(`domain` is currently a `string` but could have been a `IDirectory` if we had L2 resources for this)

Reply:
OK.